services/weatherAPI/weather_service.py

```diff
@@ -1,40 +1,82 @@
 import requests
+from abc import ABC, abstractmethod
+import urllib
+
 from services.ApiError import ApiError
 from util import Singleton
-from abc import ABC, abstractmethod
 from services.preferences import PrefService
 
 
+
 class WeatherAdapterModule(ABC):
     @abstractmethod
-    def update_current_weather_by_city(self, city:str):
+    def get_current_weather_by_city(self, city:str):
+        pass
+
+    @abstractmethod
+    def get_weather_forecast_by_city(self, city:str):
+        pass
+
+    @abstractmethod
+    def get_current_weather_by_coordinates(self, lat, lon):
         pass
 
     @abstractmethod
-    def update_weather_forecast_by_city(self, city:str):
+    def get_weather_forecast_by_coordinates(self, lat, lon):
         pass
 
 @Singleton
 class WeatherAdapterRemote(WeatherAdapterModule):
-    API_TOKEN = PrefService().get_specific_pref('openWeatherMapAPIKey')
 
-    def update_current_weather_by_city(self, city):
-        req = 'https://api.openweathermap.org/data/2.5/weather?q=' +  city + '&units=metric&appid=' + self.API_TOKEN
-        resp = requests.get(req)
-        if resp.status_code != 200:
-            raise ApiError('GET /tasks/ {}'.format(resp.status_code))
-            print('Error')
-        return resp.json()
+    def __init__(self):
+        self.API_TOKEN = PrefService().get_specific_pref('openWeatherMapAPIKey')
+        self.base_url = 'https://api.openweathermap.org/data/2.5'
+        self.base_params = {
+            'units': 'metric',
+            'appid': self.API_TOKEN
+        }
 
+    def request(self, url, params):
+        url = self.base_url + url
+        params = {**params, **self.base_params}
+        req = f'{url}?{urllib.parse.urlencode(params)}'
```

Comment:
This line is important. Living in city that consists of multiple words I failed to get the weather because you had just concatenated the city name with the URL. Spaces need to be encoded (`%20`) for it to work.